plugins/outputs/influxdb/http.go

```diff
@@ -231,11 +228,19 @@ func (c *httpClient) CreateDatabase(ctx context.Context, database string) error
 		}
 	}
 
-	// Even with a 200 response there can be an error
+	// Even with a 200 status code there can be an error in the response body.
+	// If there is also no error string then the operation was successful.
 	if resp.StatusCode == http.StatusOK && queryResp.Error() == "" {
+		c.createDatabaseExecuted[database] = true
 		return nil
 	}
 
+	// Don't attempt to recreate the database after a 403 Forbidden error.
+	// This behavior exists only to maintain backwards compatiblity.
+	if resp.StatusCode == http.StatusForbidden {
```

Comment:
shouldn't this be the other way around, we only set true if it's the specific result code that means we already have a database? Do we know what the actual code is? `400 Bad Request` or something else?

Reply:
I think it is correct.  It might help if I mention the current behavior:
- When writing to a single statically configured database, `create database` was called once and regardless of the error returned never again unless a `database not found` error is received during a write.
- When writing to dynamically configured databases via `database_tag`, `create database` was called during every write until it succeeded at least once.  If a `database not found` error is received it is also retried.

With this PR both paths use the same method:
- `Create database` is called once per database, if it fails with any error other than forbidden it will be retried the next write.  If a `database not found` error is returned it will also be retried.

Reply:
We only know the database exists if we receive a 200 status.

Reply:
Okay, I would have probably not tried creating the database unless I specifically received a `database not found`, but it sounds like this is closer to the original behavior. Will resolve and approve.